src/plugin/ApiMap.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\plugin;
+
+use function get_class;
+use function is_string;
+use InvalidArgumentException;
+use ReflectionClass;
+use RuntimeException;
+use pocketmine\utils\Utils;
+
+/**
+ * Alows different modules to expose APIs using types.
+ */
+final class ApiMap{
+	/**
+	 * @var ApiMapEntry[]
+	 * @phpstan-var array<class-string, ApiMapEntry>
+	 */
+	private $apiMap = [];
+
+	/**
+	 * @see Server#provideApi
```

Comment:
```suggestion
	 * @see Server::provideApi()
```
Also, this won't be properly resolved because `Server` isn't imported.